lessons/basics/documentation.md

```diff
@@ -0,0 +1,331 @@
+```
+
+Notice how you can use markup within out documentation and the terminal will render it? Apart from really being cool and a novel addition to Elixir's vast ecosystem, it gets much more interesting when we look at ExDoc to generate HTML documentation on the fly.
+
+## ExDoc
+
+ExDoc is an official Elixir project that **produces HTML (HyperText Markup Language and online documentation for Elixir projects** that can be found on [GitHub](https://github.com/elixir-lang/ex_doc). First lets create a Mix project for our application:
+
+```bash
+~/$ mix new greet_everyone                 
+
+* creating README.md
+* creating .gitignore
+* creating mix.exs
+* creating config
+* creating config/config.exs
+* creating lib
+* creating lib/greet_everyone.ex
+* creating test
+* creating test/test_helper.exs
+* creating test/greet_everyone_test.exs
+
+Your Mix project was created successfully.
+You can use "mix" to compile it, test it, and more:
+
+    cd greet_everyone
+    mix test
+
+Run "mix help" for more commands.
+~/$ cd greet_everyone
+
+```
+
+Now copy and paste the code from the `@doc` annotator lesson into a file called `lib/greeter.ex` and make sure everything is still working from the command line. Now that we are working within a Mix project we need to start IEx a little differently using the `iex -S mix` sequence:
+
+```bash
+greet_everyone/$ ~ iex -S mix # loads the mix project into IEx
+
+Interactive Elixir (1.2.0) - press Ctrl+C to exit (type h() ENTER for help)
+
+iex(1)> h Greeter.hello
+
+                def hello(name)                                 
+
+Prints a hello message
+
+Parameters
+
+  • name (string): The name of the person.
+
+Examples
+
+  iex> Greeter.hello("Sean") 
+  "Hello, Sean"
+
+  iex> Greeter.hello("pete") 
+  "Hello, pete"
+
+iex(2)> 
+```
+
+### Installing
+
+Assuming all being well, and we're seeing the output above suggests that we are ready to set up ExDoc. Within our `mix.exs` file add the two required dependencies to get started; `:earmark` and `:ex_doc`.
+
+```elixir
+  def deps do
+    [{:earmark, "~> 0.1", only: :dev},
+    {:ex_doc, "~> 0.11", only: :dev}]
+  end
+```
+
+We specify the `only: :dev` key-value pair as we don't want to download and compile these dependencies in a production environment. But why Earmark? Earmark is an markdown parser for the elixir programming language that ExDoc utilises to turn our documentation within `@moduledoc` and `@doc` to beautiful looking HTML. 
+
+It is worth noting at this point, that you are not forced to use Earmark. You can change the markup tool to others such as Pandoc or Hoedown; however you will need to do a little more configuration which you can read about [here](https://github.com/elixir-lang/ex_doc#changing-the-markdown-tool). For this tutorial, we'll just stick with Earmark.
+
+### Generating Documentation
+
+Carrying on, from the command line run the following two commands:
+
+```bash
+greet_everyone/$ ~ mix deps.get # gets ExDoc + Earmark.
```

Comment:
I've been using `$` to denote when in the shell versus `iex>` for IEx; I've _tried_ to be consistent throughout the lessons.  

Any objections to changing these lines to something like, to match the other lessons?

```shell
$ mix deps.get
$ mix docs
```